contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/getters.js

```diff
@@ -46,6 +47,37 @@ export function getContentNodeChildren(state) {
   };
 }
 
+/**
+ * Returns an array of all parent nodes of a node.
+ * Parent nodes are sorted from the immmediate parent
+ * to the most distant parent.
+ */
+export function getContentNodeParents(state) {
```

Comment:
Can also use https://github.com/learningequality/studio/blob/vue-refactor/contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/getters.js#L49

Reply:
Good thanks, I'd like to use this one too.

@jayoshih

1. Is there a reason for a node itself to be included in its parents or is it fine to update
`n => n.sort_order <= node.sort_order && n.rght >= node.rght`
to
`n => n.sort_order < node.sort_order && n.rght > node.rght`
?
I don't have a strong opinion - it's rather that I would expect the node itself to not be there. Though I can also remove it if needed in a view so no problem to work with it in that way too.

2. I was thinking about removing `.map(node => node.id)` so whole nodes would be returned by getter (and I would do the same for `getContentNodeChildren` for the sake of consistency). Are you okay with that if I update all places where these getters are called? For my use-case, I need to have all data available and getter already works with them actually so it might be more performant to not map values to ids and than back to nodes in views.

3. Can I understand right that channels are not included as parents in our tree data if we traverse the tree using left/right values? Now when I think about it it makes sense to me to treat channels in a different way, just to be sure that I get it.

Reply:
1. Could you add an `includeSelf` option to the code to go between the two? As it stands, most (if not all) of the breadcrumbs include the current node as well, so I implemented it this way to avoid having to concat the current node every time we want to use breadcrumbs

2. That works for me, although not sure if @rtibbles had a reason for converting them to ids on the children getter? Want to make sure the two match

3. Yes, we do want to have channels/root nodes as part of the tree data as it's causing a lot of issues across the board (unable to create items on the root node, root node not accessible in breadcrumbs, etc.)

Reply:
1. Of course

Reply:
Ah I missed that. Makes sense, thank you.

Reply:
@rtibbles 
2. I've just checked it and found out that getters actually work with the whole ContentNode objects. They are based on `getContentNode` functions that reaches to `state.contentNodesMap`.
Anyway, I will leave as is and just map it for now as I am not 100% how it was meant. We can optimize it over time.

Reply:
@jayoshih 
1. Leaving as is, I was thinking that for the most common use cases we need the node itself indeed so it will be simpler.

Reply:
I'll update this together with connecting to new breadcrumbs component - there are some overlapping areas so it will be easier

Reply:
Follow-up here https://github.com/learningequality/studio/issues/1815